pandas/core/ops/__init__.py

```diff
@@ -474,11 +475,57 @@ def dispatch_to_series(left, right, func, str_rep=None, axis=None):
     """
     # Note: we use iloc to access columns for compat with cases
     #       with non-unique columns.
+    eval_kwargs = eval_kwargs or {}
+
     import pandas.core.computation.expressions as expressions
 
     right = lib.item_from_zerodim(right)
+
     if lib.is_scalar(right) or np.ndim(right) == 0:
 
+        new_blocks = []
+        mgr = left._data
+        for blk in mgr.blocks:
+            # Reshape for EA Block
+            blk_vals = blk.values
+            if hasattr(blk_vals, "reshape"):
+                # ndarray, DTA/TDA/PA
+                blk_vals = blk_vals.reshape(blk.shape)
+                blk_vals = blk_vals.T
+
+            new_vals = array_op(blk_vals, right, func, str_rep, eval_kwargs)
+
+            # Reshape for EA Block
+            if is_extension_array_dtype(new_vals.dtype):
+                from pandas.core.internals.blocks import make_block
+
+                if hasattr(new_vals, "reshape"):
+                    # ndarray, DTA/TDA/PA
+                    new_vals = new_vals.reshape(blk.shape[::-1])
+                    assert new_vals.shape[-1] == len(blk.mgr_locs)
+                    for i in range(new_vals.shape[-1]):
+                        nb = make_block(new_vals[..., i], placement=[blk.mgr_locs[i]])
+                        new_blocks.append(nb)
+                else:
+                    # Categorical, IntegerArray
+                    assert len(blk.mgr_locs) == 1
+                    assert new_vals.shape == (blk.shape[-1],)
+                    nb = make_block(new_vals, placement=blk.mgr_locs, ndim=2)
+                    new_blocks.append(nb)
+            elif blk.values.ndim == 1:
```

Comment:
I'm curious: what hits this case? An op on an EA that returns an ndarray? Say `DataFrame[Categorical] == 0`?

Reply:
Exactly.  `Block[Categorical].values == 0` needs to become a 2D ndarray[bool] block before we're done.